[2025-05-26T08:43:38.799+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-05-26T08:43:38.810+0000] {taskinstance.py:2603} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: x_videos_scraper_dag.get_links_task manual__2025-05-26T08:43:29.924092+00:00 [queued]>
[2025-05-26T08:43:38.816+0000] {taskinstance.py:2603} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: x_videos_scraper_dag.get_links_task manual__2025-05-26T08:43:29.924092+00:00 [queued]>
[2025-05-26T08:43:38.816+0000] {taskinstance.py:2856} INFO - Starting attempt 1 of 6
[2025-05-26T08:43:38.824+0000] {taskinstance.py:2879} INFO - Executing <Task(PythonOperator): get_links_task> on 2025-05-26 08:43:29.924092+00:00
[2025-05-26T08:43:38.827+0000] {standard_task_runner.py:72} INFO - Started process 323411 to run task
[2025-05-26T08:43:38.830+0000] {standard_task_runner.py:104} INFO - Running: ['airflow', 'tasks', 'run', 'x_videos_scraper_dag', 'get_links_task', 'manual__2025-05-26T08:43:29.924092+00:00', '--job-id', '6804', '--raw', '--subdir', 'DAGS_FOLDER/pipeline_x.py', '--cfg-path', '/tmp/tmp4m9v66mj']
[2025-05-26T08:43:38.831+0000] {standard_task_runner.py:105} INFO - Job 6804: Subtask get_links_task
[2025-05-26T08:43:38.869+0000] {task_command.py:467} INFO - Running <TaskInstance: x_videos_scraper_dag.get_links_task manual__2025-05-26T08:43:29.924092+00:00 [running]> on host f541e6bad3a5
[2025-05-26T08:43:38.924+0000] {taskinstance.py:3122} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='x_videos_scraper_dag' AIRFLOW_CTX_TASK_ID='get_links_task' AIRFLOW_CTX_EXECUTION_DATE='2025-05-26T08:43:29.924092+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-05-26T08:43:29.924092+00:00'
[2025-05-26T08:43:38.925+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-05-26T08:43:51.293+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926913878014701857/video/1
[2025-05-26T08:43:51.299+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926750119032152371/video/1
[2025-05-26T08:43:51.302+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926747602542002315/video/1
[2025-05-26T08:43:51.306+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926745087414829232/video/1
[2025-05-26T08:43:51.310+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926709942288855300/video/1
[2025-05-26T08:43:51.324+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926698131439140936/video/1
[2025-05-26T08:43:51.349+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926678327118479510/video/1
[2025-05-26T08:43:51.364+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926660027869245813/video/1
[2025-05-26T08:43:51.394+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926352528414744716/video/1
[2025-05-26T08:43:51.403+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926231736901407125/video/1
[2025-05-26T08:43:51.407+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1926216780638192037/video/1
[2025-05-26T08:43:51.421+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1925944832612012205/video/1
[2025-05-26T08:43:51.424+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1925893335006195900/video/1
[2025-05-26T08:43:51.428+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1925892968231109016/video/1
[2025-05-26T08:43:51.438+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1925861711950512549/video/1
[2025-05-26T08:43:51.452+0000] {logging_mixin.py:190} INFO - https://x.com/ManCity/status/1925653957633855876/video/1
[2025-05-26T08:43:51.500+0000] {logging_mixin.py:190} INFO - Remaining new videos: 16
[2025-05-26T08:43:51.575+0000] {taskinstance.py:3301} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 762, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/dags/pipeline_x.py", line 21, in run_x_videos_scraper
    return x_videos_scraper(
  File "/opt/airflow/dags/tasks/x_videos_scraper.py", line 59, in x_videos_scraper
    hrefs = hrefs[:scrolls]  # Giới hạn số lượng video mới lấy về
NameError: name 'scrolls' is not defined
[2025-05-26T08:43:51.587+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=x_videos_scraper_dag, task_id=get_links_task, run_id=manual__2025-05-26T08:43:29.924092+00:00, execution_date=20250526T084329, start_date=20250526T084338, end_date=20250526T084351
[2025-05-26T08:43:51.587+0000] {taskinstance.py:1554} INFO - Executing <lambda> callback
[2025-05-26T08:43:51.587+0000] {pipeline_x.py:40} INFO - Retrying DAG
[2025-05-26T08:43:51.600+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-05-26T08:43:51.600+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 6804 for task get_links_task (name 'scrolls' is not defined; 323411)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 2995, in _run_raw_task
    return _run_raw_task(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3149, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 3173, in _execute_task
    return _execute_task(self, context, task_orig)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 762, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
  File "/home/airflow/.local/lib/python3.10/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
  File "/opt/airflow/dags/pipeline_x.py", line 21, in run_x_videos_scraper
    return x_videos_scraper(
  File "/opt/airflow/dags/tasks/x_videos_scraper.py", line 59, in x_videos_scraper
    hrefs = hrefs[:scrolls]  # Giới hạn số lượng video mới lấy về
NameError: name 'scrolls' is not defined
[2025-05-26T08:43:51.632+0000] {local_task_job_runner.py:261} INFO - Task exited with return code 1
[2025-05-26T08:43:51.657+0000] {taskinstance.py:3891} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-05-26T08:43:51.659+0000] {local_task_job_runner.py:240} INFO - ::endgroup::
